django_mongodb/compiler.py

```diff
@@ -154,7 +154,7 @@ def get_columns(self):
             self.get_default_columns(select_mask) if self.query.default_cols else self.query.select
         )
         return tuple((column.target.column, column) for column in columns) + tuple(
-            self.query.annotations.items()
+            self.query.annotation_select.items()
```

Comment:
The compilers also have extra fields (I'm not sure when they are used, but they are there). For now, I think we can continue with this approach; it will probably change in the future, but it won't be a big change.

How much more does annotation_select cover?

Reply:
`annotation_select` has fewer fields than `annotations`: it doesn't include the annotations of [`QuerySet.alias()`](https://docs.djangoproject.com/en/dev/ref/models/querysets/#alias). The fields of `alias()` aren't supposed to appear in the `SELECT` clause and for us that means `$project`. `test_basic_alias_f_transform_annotation` failed because the list of selected fields here didn't match [what Django expects](https://github.com/django/django/blob/e99187e5c94516ee35f37cc41a36d906b395808d/django/db/models/query.py#L129-L131) so the wrong values were loaded into the wrong fields.